brave/brave-core
# Uphold wallet

Resolves https://github.com/brave/brave-browser/issues/5005
Resolves https://github.com/brave/brave-browser/issues/5004
Resolves https://github.com/brave/brave-browser/issues/4774
Resolves https://github.com/brave/brave-browser/issues/4776
Resolves https://github.com/brave/brave-browser/issues/4775
Resolves https://github.com/brave/brave-browser/issues/4779

UI PR: https://github.com/brave/brave-ui/pull/497

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:
1. Linking
- clean profile
- enable rewards
- click on Verify wallet
- make sure that you see on boarding
- make sure that you are redirected to uphold side
- sign in or create account (try with new profile, existing profile and existing kyc'd profile)
- make sure that you are redirected back to rewards page
  - if you signed in with KYC'd account you should see green button saying verified
  - if you signed in with a regular account you should be redirected to uphold kyc page

2. Contribution
- clean profile
- enable rewards
- link to kyc'd wallet
- tip one verified publisher
- make sure that in your uphold account you see two transactions (5% that goes to brave and 95% that went to the publisher)

3. Claim
- clean profile
- enable rewards
- recover wallet that was self funded
- link to kyc'd wallet
- make sure that user funds from anon wallet are transferred to uphold account that you connected

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
